beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3031,16 +3022,51 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             validator_graffiti,
             ProduceBlockVerification::VerifyRandao,
         )
+        .await
     }
 
     /// Same as `produce_block` but allowing for configuration of RANDAO-verification.
-    pub fn produce_block_with_verification<Payload: ExecPayload<T::EthSpec>>(
+    pub async fn produce_block_with_verification<Payload: ExecPayload<T::EthSpec>>(
         self: &Arc<Self>,
         randao_reveal: Signature,
         slot: Slot,
         validator_graffiti: Option<Graffiti>,
         verification: ProduceBlockVerification,
     ) -> Result<BeaconBlockAndState<T::EthSpec, Payload>, BlockProductionError> {
+        // Part 1/2 (blocking)
+        //
+        // Load the parent state from disk.
+        let chain = self.clone();
+        let (state, state_root_opt) = self
+            .task_executor
+            .spawn_blocking_handle(
```

Comment:
I also attempted this, but the containing function returns a `BlockProductionError` rather than a `BeaconChainError` so the `BeaconChain::spawn_blocking_handle` method isn't so effective unfortunately.